src/types/presets.ts

```diff
@@ -65,4 +67,13 @@ export interface PresetOptions {
       maxDuration: number;
     };
   };
+  firebase: {
```

Comment:
What would be the difference with runtimeConfig? I see other presets using `useRuntimeConfig()` but none using `useAppConfig()`

Reply:
@posva I guess you're referencing this line where we add the config to the appConfig. The only reason we do this is to be able to reference it in the entry file here:
https://github.com/unjs/nitro/pull/1142/files#diff-89af85ba779addbc7c61eaecf298b36da9e6d1d5753f25ff55416ef9f50d833eR8

It's all put under a key that isn't intended to be used by the end user. This was implemented on @Hebilicious suggestion here:
https://github.com/unjs/nitro/pull/1142#discussion_r1250797577

We took the options out of runtimeConfig to put them along side other preset config like vercel:
https://github.com/unjs/nitro/pull/1142#issuecomment-1512736670

Nitro options seem like the best place to put these as you are configuring the nitro build at this point rather than runtime options. Then I guess putting them either in app config or runtime config doesn't really matter as it's just a way to get those options into the entry file?

Happy to change it though if it's better practice @Hebilicious @pi0 ?

Reply:
Thanks! This really helps to follow the flow of the PR!

Reply:
@Hebilicious any thoughts on this?

Reply:
> @Hebilicious any thoughts on this?

Everything looked good to me the last time I reviewed it, I believe @posva is adding automated tests in nitro deploys and @pi will be in charge of merging.